src/v2/guide/components.md

```diff
@@ -963,68 +963,57 @@ The content distribution API is a very useful mechanism when designing component
-
-In a child component, pass data into a slot as if you are passing props to a component:
-
-``` html
-<div class="child">
-  <slot text="hello from child"></slot>
-</div>
-```
-
-In the parent, a `<template>` element with a special attribute `slot-scope` must exist, indicating that it is a template for a scoped slot. The value of `slot-scope` will be used as the name of a temporary variable that holds the props object passed from the child:
-
-``` html
-<div class="parent">
-  <child>
-    <template slot-scope="props">
-      <span>hello from parent</span>
-      <span>{{ props.text }}</span>
-    </template>
-  </child>
-</div>
-```
-
-If we render the above, the output will be:
+Scoped slots allow the definition of dynamic slot templates, using data passed from a child component. For example, in this hypothetical `<review-list>` component:
 
 ``` html
-<div class="parent">
-  <div class="child">
-    <span>hello from parent</span>
-    <span>hello from child</span>
-  </div>
-</div>
+<ul>
+  <li v-for="review in reviews">
+    <slot :review="review"></slot>
+  </li>
+</ul>
 ```
 
-> In 2.5.0+, `slot-scope` is no longer limited to `<template>` and can be used on any element or component.
+We're passing `review` to the slot, just like props to a component.
 
-A more typical use case for scoped slots would be a list component that allows the component consumer to customize how each item in the list should be rendered:
+Now when we use this component, we can use a `slot-scope` attribute to access these props, binding them to an object called `slotProps` in the example below:
 
 ``` html
-<my-awesome-list :items="items">
-  <!-- scoped slot can be named too -->
-  <li
-    slot="item"
-    slot-scope="props"
-    class="my-fancy-item">
-    {{ props.text }}
-  </li>
-</my-awesome-list>
+<h1>Popular reviews for our restaurant</h1>
+<review-list :reviews="restaurantReviews">
+  <template slot-scope="slotProps">
+    <h3>{{ slotProps.review.title }}</h3>
+    <p class="byline">
+      {{ slotProps.review.author }}
+    </p>
+    <p>{{ slotProps.review.content }}</p>
+  </template>
+</review-list>
 ```
 
-And the template for the list component:
+The rendered output for this example might be:
 
 ``` html
+<h1>Popular reviews for our restaurant</h1>
 <ul>
-  <slot name="item"
-    v-for="item in items"
-    :text="item.text">
-    <!-- fallback content here -->
-  </slot>
+  <li>
+    <h3>Very good food</h3>
+    <p class="byline">
+      Gregor Clegane
+    </p>
+    <p>Try the sushi. It's my favorite.</p>
+  </li>
+  <li>
+    <h3>So many choices!</h3>
+    <p class="byline">
+      Samwise Gamgee
+    </p>
+    <p>Normally, only one or two things look good to me, but here, everything is delicious!</p>
+  </li>
 </ul>
 ```
 
+> Before 2.5.0, the `slot-scope` attribute was limited to `<template>` elements, but can now be used with any element or component, on or inside a slot.
```

Comment:
> **on** or inside a slot.

I don't fully understand this. That means the following code work?

```html
<review-list
  :reviews="restaurantReviews"
  slot-scope="slotProps">
  <h3>{{ slotProps.review.title }}</h3>
  <p class="byline">
    {{ slotProps.review.author }}
  </p>
  <p>{{ slotProps.review.content }}</p>
</review-list>
```

Reply:
No, because `<review-list>` is not on or inside a slot (it's not inside itself). Instead, the default slot starts _inside_ the component. This is one of the areas of confusion I was trying to make clearer and have apparently failed. 😅  What do you think could make it clearer?

Reply:
If the doc contained this, I still would have misunderstood. Like I said though, I missed the mention of "child elements" earlier in the doc so maybe that's just my own fault. "on or inside a slot" doesn't clarify it further from my point of view. If there's only one slot, I would've thought the wrapping tags were an acceptable place (i.e. on the slot).

Maybe making it "child element or component" would be best? You could go further and a HTML comment to the example?

Reply:
[L1015](https://github.com/vuejs/vuejs.org/pull/1312/files?diff=split#diff-044f0befe8adfe59792a31b1763e3cc4R1015) is confusing. Maybe providing an inline example would help. 

@chrisvfritz this PR should be run by someone new to vue.